Source/WebCore/rendering/RenderTheme.cpp

```diff
@@ -458,12 +460,20 @@ bool RenderTheme::paint(const RenderBox& box, ControlStates& controlStates, cons
     }
     if (paintInfo.context().paintingDisabled())
         return false;
+    
+    ControlPart part = box.style().effectiveAppearance();
+    IntRect integralSnappedRect = snappedIntRect(rect);
+
+    // Temporarily move this call above the canPaint check to allow
```

Comment:
This says "Temporarily" but is there a long-term fix?

Reply:
Maybe a good solution to this is to explicitly allow painting of certain `ControlPart` in `canPaint` without a platform context.